service/share/testing.go

```diff
@@ -74,63 +60,133 @@ func RandShares(t *testing.T, n int) []Share {
 	return ipld.RandShares(t, n)
 }
 
-type DAGNet struct {
-	ctx context.Context
-	t   *testing.T
-	net mocknet.Mocknet
+type node struct {
+	*Service
+	blockservice.BlockService
+	host.Host
+}
+
+type dagNet struct {
+	ctx   context.Context
+	t     *testing.T
+	net   mocknet.Mocknet
+	nodes []*node
 }
 
-func NewDAGNet(ctx context.Context, t *testing.T) *DAGNet {
-	return &DAGNet{
+func NewTestDAGNet(ctx context.Context, t *testing.T) *dagNet { //nolint:revive
+	return &dagNet{
 		ctx: ctx,
 		t:   t,
 		net: mocknet.New(ctx),
 	}
 }
 
-func (dn *DAGNet) RandLightService(n int) (*Service, *Root) {
-	bServ, root := dn.RandDAG(n)
-	return NewService(bServ, NewLightAvailability(bServ)), root
+func (dn *dagNet) RandLightNode(n int) (*node, *Root) {
+	nd := dn.LightNode()
+	return nd, RandFillDAG(dn.t, n, nd.BlockService)
 }
 
-func (dn *DAGNet) RandFullService(n int) (*Service, *Root) {
-	bServ, root := dn.RandDAG(n)
-	return NewService(bServ, NewFullAvailability(bServ)), root
+func (dn *dagNet) RandFullNode(n int) (*node, *Root) {
```

Comment:
While you're here, could you briefly document these? I wasn't actually sure what the parameter was until I read the implementation.